src/api/cached_op_api.cc

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file cached_op_api.cc
+ * \brief The API of function to invoke CachedOp in src/imperative/cached_op.cc
+ */
+#include <mxnet/api_registry.h>
+#include <mxnet/runtime/packed_func.h>
+#include <mxnet/runtime/container_ext.h>
+#include "../imperative/cached_op.h"
+#include "../imperative/cached_op_threadsafe.h"
+
+namespace mxnet {
+
+MXNET_REGISTER_GLOBAL("cached_op.invoke")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  CachedOpPtr op_shared = *static_cast<CachedOpPtr*>(
+    static_cast<CachedOpHandle>(static_cast<void*>(args[0])));
```

Comment:
Why 3 nested static_cast? Note that `typedef void *CachedOpHandle;`
This question also applies to the other locations where you do nested static_cast.

Reply:
Thanks for pointing it out. I have updated the type checking by removing the nested static_cast. Actually, MXNetArgs cannot be directly cast to some special pointers. We can either cast them to void* or access the v_handle of MXNetArgs.value() first and then cast to special pointers.